force-app/main/default/classes/NPSP_Address.cls

```diff
@@ -341,6 +341,15 @@ public inherited sharing class NPSP_Address implements IAddress {
         address.Geolocation__Latitude__s = (Decimal) getSObjectField(sobjSrc, fieldPrefixSrc, 'Latitude');
         address.Geolocation__Longitude__s = (Decimal) getSObjectField(sobjSrc, fieldPrefixSrc, 'Longitude');
 
+        if (orgConfig.isStateCountryPicklistsEnabled()) {
```

Comment:
It seems a little strange that we allow the MailingState and MailingCountry on the address to potentially have a value copied to it in the above block, and then override that value with setting another value in this `if` block. And with the country code null check below, that would mean the country code wouldn't be copied to the address if it already has a value, even if it's not the country code

Reply:
Want to take another look now that we've reverted this logic?